GooglePlayInstant/Tests/Editor/QuickDeploy/HttpRequestHelperTest.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using GooglePlayInstant.Editor.QuickDeploy;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace GooglePlayInstant.Tests.Editor.QuickDeploy
+{
+    /// <summary>
+    /// Tests for HttpRequestHelper class methods.
+    /// </summary>
+    [TestFixture]
+    public class HttpRequestHelperTest
+    {
+        /*
+         * Testing strategy:
+         *     - Test helper methods that manipulate data that is fed into instantiation of WWW instances.
+         *     - Exclude methods that instantiate the WWW class from unit tests to avoid sending HTTP requests.
+         */
+
+        [Test]
+        public void TestGetWwwForm()
+        {
+            var postParams =
+                HttpRequestHelperTestHelper.GetKeyValueDict(new[] {"a", "b", "c", "d"}, key => string.Concat(key, key));
+            var form = HttpRequestHelper.GetWwwForm(postParams);
+            Assert.AreEqual(postParams,
+                HttpRequestHelperTestHelper.GetDictFromUrlQuery(string.Format("?{0}",
+                    Encoding.UTF8.GetString(form.data))));
+        }
+
+        [Test]
+        public void TestGetEndpointWithGetParams()
+        {
+            var getParams =
+                HttpRequestHelperTestHelper.GetKeyValueDict(new[] {"a", "b", "c", "d"}, key => string.Concat(key, key));
+            var endPointWithQuery = HttpRequestHelper.GetEndpointWithGetParams("http://localhost:5000", getParams);
+            Assert.AreEqual(getParams,
+                HttpRequestHelperTestHelper.GetDictFromUrlQuery(new Uri(endPointWithQuery).Query));
+        }
+
+
+        [Test]
+        public void TestGetCombinedDictionary()
```

Comment:
Add tests for null or empty parameters. These can be very simple tests and arguably can be a separate method.

Reply:
Addressed.